src/core/mesh/qgsmeshrenderersettings.h

```diff
@@ -440,6 +440,75 @@ class CORE_EXPORT QgsMeshRendererVectorSettings
     QgsMeshRendererVectorStreamlineSettings mStreamLinesSettings;
 };
 
+/**
+ * \ingroup core
+ *
+ * Represents mesh renderer settings for averaging method QgsMeshSingleLevelAverageMethod
+ *
+ * \note The API is considered EXPERIMENTAL and can be changed without a notice
+ *
+ * \since QGIS 3.12
+ */
+class CORE_EXPORT QgsMeshRendererAveragingSingleVerticalLayerSettings
+{
+  public:
+
+    //! Returns index of vertical layer to show (indexed from 0, from top layer)
+    int verticalLayer() const;
+    //! Sets index of vertical layer to show (indexed from 0, from top layer)
+    void setVerticalLayer( int verticalLayer );
+
+    //! Writes configuration to a new DOM element
+    QDomElement writeXml( QDomDocument &doc ) const;
+    //! Reads configuration from the given DOM element
+    void readXml( const QDomElement &elem );
+  private:
+    int mVerticalLayer = 0;
+};
+
+/**
+ * \ingroup core
+ *
+ * Represents mesh renderer settings for averaging method
+ *
+ * \note The API is considered EXPERIMENTAL and can be changed without a notice
+ *
+ * \since QGIS 3.12
+ */
+class CORE_EXPORT QgsMeshRenderer3dAveragingSettings
+{
+  public:
+
+    /**
+     * Defines the method of averaging
+     */
+    enum Method
+    {
+      SingleVerticalLayer = 0, //!< QgsMeshSingleLevelAverageMethod
+    };
+
+    //! Returns averaging method
+    Method averagingMethod() const;
+
+    //! Sets averaging method
+    void setAveragingMethod( const Method &averagingMethod );
+
+    //! Returns settings for QgsMeshSingleLevelAverageMethod
+    QgsMeshRendererAveragingSingleVerticalLayerSettings singleVerticalLayerSettings() const;
+
+    //! Sets settings for QgsMeshSingleLevelAverageMethod
+    void setSingleVerticalLayerSettings( const QgsMeshRendererAveragingSingleVerticalLayerSettings &singleVerticalLayerSettings );
+
+    //! Writes configuration to a new DOM element
+    QDomElement writeXml( QDomDocument &doc ) const;
+    //! Reads configuration from the given DOM element
+    void readXml( const QDomElement &elem );
+
+  private:
+    Method mAveragingMethod = SingleVerticalLayer;
+    QgsMeshRendererAveragingSingleVerticalLayerSettings mSingleVerticalLayerSettings;
+};
+
```

Comment:
It would be good to get rid of these classes and just use directly the averaging method classes for mesh renderer settings as well, to avoid duplication